DEVEX-2305 Add dxpy dependencies cross-platform testing suite (#1027)
dnanexus/dx-toolkit: build/dependencies_sanity_tests/README.md

> # dxpy dependencies sanity tests

Reviewer: ```suggestion
# dxpy dependencies cross platform tests
```

## Usage

### Linux

1. Install Python and Docker: `sudo apt install --yes python3 docker.io`
2. Install Docker Python: `python3 -m pip install docker`
3. Run tests: `./run_linux.py -t <token> -d <dx-toolkit path> ...`

### Windows

1. Run PowerShell as an administrator
2. Prepare environment (Pythons, etc.): `powershell.exe -ExecutionPolicy Unrestricted windows\prepare.ps1`
3. Run tests: `python3.11 run_windows.py -t <token> -d <dx-toolkit path> ...`

### MacOS

1. Install dependencies: `bash macos/prepare.sh`
2. Run tests: `./run_macos.sh -t <token> -d <dx-toolkit path> ...`

## Where are libraries used

### argcomplete

Argument completion for `dx` commands in Unix shells. Tested usign `pexpect`.

### colorama

Colorized terminal output on Windows. Called directly in `dx` and `dx-app-wizard` scripts. Don't know how to test automatically.

### cryptography

Seems to be unused and present for historical reasons. In dxpy, the library is NOT used anywhere. 

> The only place I could found is `urllib3` contrib module `PyOpenSSL` (https://urllib3.readthedocs.io/en/stable/reference/contrib/pyopenssl.html) and it is required for certificate validation (using packace `urllib3[secure]`) in Python 2: https://urllib3.readthedocs.io/en/stable/user-guide.html#certificate-verification-in-python-2 and https://urllib3.readthedocs.io/en/stable/user-guide.html#certificate-verification.

Reviewer: ```suggestion
The only place where it seems to be used is with the `urllib3` contrib module `PyOpenSSL` (https://urllib3.readthedocs.io/en/stable/reference/contrib/pyopenssl.html) and it is required for certificate validation (using packace `urllib3[secure]`) in Python 2: https://urllib3.readthedocs.io/en/stable/user-guide.html#certificate-verification-in-python-2 and https://urllib3.readthedocs.io/en/stable/user-guide.html#certificate-verification.
```
